File: agents/java-agent/src/main/java/com/hmdev/messaging/agent/api/impl/UdpEnvelope.java
package com.hmdev.messaging.agent.api.impl;

import com.fasterxml.jackson.databind.JsonNode;
import lombok.Getter;
import lombok.Setter;

/**
 * Simple model for UDP envelope messages sent to the service UDP listener.
 * Structure: { "action": "push" | "pull", "payload": { ... }, "requestId": "..." }
 */
@Setter
@Getter
public class UdpEnvelope {
    private String action;
    private JsonNode payload;
    private String requestId;

    public UdpEnvelope() { }

    public UdpEnvelope(String action, JsonNode payload) {
        this.action = action;
        this.payload = payload;
    }

    public UdpEnvelope(String action, JsonNode payload, String requestId) {
        this.action = action;
        this.payload = payload;
        this.requestId = requestId;
    }

}

File: agents/python-agent/hmdev/messaging/agent/api/impl/udp_envelope.py
# Simple UDP envelope model to mirror Java's UdpEnvelope
from __future__ import annotations
from dataclasses import dataclass
from typing import Any, Dict, Optional


@dataclass
class UdpEnvelope:
    action: str
    payload: Dict[str, Any]
    requestId: Optional[str] = None

    def to_dict(self) -> Dict[str, Any]:
        data = {
            "action": self.action,
            "payload": self.payload,
        }
        if self.requestId:
            data["requestId"] = self.requestId
        return data

